places/views.py

```diff
@@ -5,35 +5,48 @@
 
 from places.models        import Place
 
+
 class PlaceListView(View):    
     def get(self, request):
-        try:
-            filter_prefixes = {
-                'menu'     : 'category__menu_id__in',
-                'category' : 'category_id__in'
-            }
+        try: 
+            menu     = request.GET.get('menu', None)
+            category = request.GET.get('category', None)
+            keyword  = request.GET.get('keyword', None)
+            order    = request.GET.get('order', None)
+        
+            if menu:
+                places = Place.objects.filter(category__menu__id = menu)
             
-            filter_set = {filter_prefixes.get(key) : value for (key, value) in dict(request.GET).items()}
+            if category:
+                places = Place.objects.filter(category__id = category)
+                
+            if keyword:
+                places = Place.objects.filter(name__contains=keyword)
+                
+            if order:
+                places = Place.objects.order_by(order)
```

Comment:
if문을 사용할 거라면 Q()객체를 사용해서 구현하는게 더 좋습니다. 왜냐하면 and 조건 뿐만 아니라 or 조건에도 유연하게 대응할 수 있기 때문이죠.

```suggestion
            order    = request.GET.get('order', 'id')
            
            q = Q()
            
            if category:
                q &= Q(category__id = category)
                
            if keyword:
                q &= Q(name__contains=keyword)       
                
            places = Place.objects.filter(q).order_by(order)
```

Reply:
위와 같이 코드를 수정해 주세요!